### DISCUSSION TITLE
Recording files disappear

### DISCUSSION THREAD

**Discussion by emersonnpe** [Ylianst/MeshCentral#6161] [General]
Dear community colleagues,

I came across a problem regarding the recordings. They are not staying and when trying to execute them minutes later it is not possible. 

Below is information about the configuration file.

"SessionRecording": {
        "onlySelectedDeaviceGroups": true,
        "filepath": "/meshcentral/gravacoes",
        "index": true,
        "maxRecordings": 50,
        "maxRecordingSizeMegabytes": 50,
        "protocols": [1,2,200] 

![image](https://github.com/Ylianst/MeshCentral/assets/75445621/f9584d38-887d-444a-a6b5-97eabfaa3500)

**Comment by si458** [comment]
this is because of a number of things

1. you have `index: true` set
so the minute you end a session, your server starts to index the files it has just created, so they arent always available and ready
and sometimes the files can be BIG in size and so it can take 10-15mins EACH to index depending on the size of the file, how long the session was, what sort of spec machine you have.
so i would recommend to set `index: false` unless u 110% want to be able to fast forward in 10 second increments when you play back the footage

2. you have set `maxRecordingSizeMegabytes` which limits the size of the file that can be created/stored,
so if a recording goes over this size, it wont save it
so again my recommend is to remove `maxRecordingSizeMegabytes`,
then do a few sessions for a length of time, and see what sort of size you get for the sessions,
sometimes i get file sizes of like 100MB!? and if you look at your 3rd entry its about 44mb in size for 28mins of recording!

3. im not sure if its a spelling error or translate error but `onlySelectedDeaviceGroups` should be `onlySelectedDeviceGroups`

p.s. you also havent said what version of meshcentral you are using, what server you run it on, how u store the footage

**Reply by emersonnpe** [reply]
Thank you for your help. 

onlySelectedDeviceGroups was actually misspelled. I've already adjusted it and am re-running the tests.

Server: Linux Debian 12
Meshcentral: 1.1.22
Storage: Recording files are stored on a separate disk as we monitor this disk in case it is possible to audit the recordings.

**Comment by emersonnpe** [comment]
Dear all, I carried out the steps above and am monitoring to see if everything is in order.
Thanks for the help. I will get back to you soon with the result.

**Reply by silversword411** [reply]
Don't forget to sponsor for the assistance provided
